### ISSUE TITLE
unable to start container named ca

### ISSUE THREAD

**Issue by jfgibbins** [lxc/lxd#3438]
The template below is mostly useful for bug reports and support questions.
Feel free to remove anything which doesn't apply to you and add more information where it makes sense.

# Required information

 * Distribution: Ubuntu
 * Distribution version: 17.04
 * The output of "lxc info" or if that fails:
config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
api_status: stable
api_version: "1.0"
auth: trusted
public: false
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  driver: lxc
  driver_version: 2.0.7
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.10.0-24-generic
  server: lxd
  server_pid: 1369
  server_version: "2.12"
  storage: zfs
  storage_version: 0.6.5.9-2

# Issue description
lxc launch ubuntu: ca -p default
fails to start container
other names such as c, cb, caa, etc, work, just "ca" won't start

# Information to attach
 lxc info ca --show-log
Name: ca
Remote: unix:/var/lib/lxd/unix.socket
Architecture: x86_64
Created: 2017/06/23 18:11 UTC
Status: Stopped
Type: persistent
Profiles: default

Log:

/var/log/lxd/lxd.log
ephemeral=false lvl=info msg="Creating container" name=ca t=2017-06-23T14:11:53-0400
ephemeral=false lvl=info msg="Created container" name=ca t=2017-06-23T14:11:53-0400

--debug
DBUG[06-23|14:15:06] Raw response: {"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":{"config":{},"api_extensions":["storage_zfs_remove_snapshots","container_host_shutdown_timeout","container_syscall_filtering","auth_pki","container_last_used_at","etag","patch","usb_devices","https_allowed_credentials","image_compression_algorithm","directory_manipulation","container_cpu_time","storage_zfs_use_refquota","storage_lvm_mount_options","network","profile_usedby","container_push","container_exec_recording","certificate_update","container_exec_signal_handling","gpu_devices","container_image_properties","migration_progress","id_map","network_firewall_filtering","network_routes","storage","file_delete","file_append","network_dhcp_expiry","storage_lvm_vg_rename","storage_lvm_thinpool_rename","network_vlan","image_create_aliases","container_stateless_copy"],"api_status":"stable","api_version":"1.0","auth":"trusted","public":false,"environment":{"addresses":[],"architectures":["x86_64","i686"],"certificate":"-----BEGIN CERTIFICATE-----\nMIIFPTCCAyWgAwIBAgIQJlQRzEuSp2/pd7slP9k6ajANBgkqhkiG9w0BAQsFADAz\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMRMwEQYDVQQDDApyb290QGx4\nZDAxMB4XDTE3MDYxOTA1NDc1NFoXDTI3MDYxNzA1NDc1NFowMzEcMBoGA1UEChMT\nbGludXhjb250YWluZXJzLm9yZzETMBEGA1UEAwwKcm9vdEBseGQwMTCCAiIwDQYJ\nKoZIhvcNAQEBBQADggIPADCCAgoCggIBAMILH6rWNyKjcm8KvWH+q8vFeOKNNZu+\nc0sDSjyLcjWyUNH6Dp4PXsB07qSMjaTxbYbq/k1P+n4LP3PMB610sbda9NNTXd0c\nuM0aQrRXCe//26h6UscjptArREQfoo/R+fEMteKh0gXtmdUA8H8G1FBwKGgXRmW3\nO9Ij8XmHJZIOQjnr3drAC+NPwyKzSw9hu2Ns7O0NEi5Aign2OApY4mdJRg377gNu\nHwYJUPMy51a337vTeNJ26EiBDALe1l97ImABFKyX+sZl9QqCzvd5uAYxcia3Ut8G\nZQR5t5nPunsDFWFfnGFRHXLEXwZoZ0Ck/6k1E5XwJ1o4x/KVOzwo3tBNPDrHMQqw\n4Ffk3TUvI7BsFSmRrVzKkZo096VvlAKzzWg6S2hg6ijPcIWFeZ1886/3IsfIpNDf\nlsdzRITK/DOlAMa9pHUd+BAwZIZcgbghCA8zqdyng8YpB8eijFHXR8ilyvMNS+mC\nWMpChehvOGKZRsSX/fKGgSaWmogAhYV9GlE6O4tmpgyf/pR41b3kLLWImvDH2GFi\nTh+k48aIqZFx8/9I5MJ6z5yGoYBf9N4gQxt/3WqJ44zGjuwm+qsexX8khzdpApQr\noAqYg48mhK5/YlZIksCk+sUQxV5FXYdsol2ISegEjpC6TQ0rlqBfOlLCJpkoueFm\n5xVo03mgcOh5AgMBAAGjTTBLMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggr\nBgEFBQcDATAMBgNVHRMBAf8EAjAAMBYGA1UdEQQPMA2CBWx4ZDAxhwQK/wAFMA0G\nCSqGSIb3DQEBCwUAA4ICAQAoo6DZHQujHfjMajgSRSyN7lUyz7vleAUU2Yk/aOx9\ng83+mcsfvRpgfpnlHVRzBWUfszgqgb9JiS79oZWiRB9eJk2dszSQFig4/x1yetkM\nJUCxcabwLhdHbk+WsrVOG9lC4psecK2ouC+rMZtZLp705fhVKB20Z7i2udD5AAwW\n+l9mKzeVddHkl/jMzYrMTvAr5XamK1v9I85lo8a6b4e/KZXFc4QNRvtCff9uvYE9\nx2ByhNvGMSk4V0asS5HJVidC7TyFtn4vuoXk96ZI3ySfXP3C7xA+77taQC59Ng/n\nVy+YH809xyb2QqOJVmphiDxE0QdpTcuWYI2lveDrBxStkxoUw8lRum3kiMwqqR18\n9kFknVhiDEGoe56uN/z8+nUSpJyybRcBNqcdsUdtwn1HC0Cn3Xa0xptJviQs//AO\ndjZgMWneq/f82bGmkh8/xReXWz7yDxfeV8MJIYFLjcgzvDtioujbiaEU5nRtS9SC\nzo7whcxl+s2pJnq0H+//KJ1xQprRxSECeKZnFOskn6b/Ux14vMwp1QLWLDhzKGg2\nnSGTORi859g3EOCbi6K5Fl0gkxVw3IFDlbbMTEHiejgK1EQdBt8+QKRSymEcTZi8\nl9WvynDXuk2rqnfnHw/K6JeV0B1/2O+w8r1a93W8Teq88CJJGgZUKHa9jnB8CVe5\nYA==\n-----END CERTIFICATE-----\n","certificate_fingerprint":"","driver":"lxc","driver_version":"2.0.7","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.10.0-24-generic","server":"lxd","server_pid":1369,"server_version":"2.12","storage":"zfs","storage_version":"0.6.5.9-2"}}}

DBUG[06-23|14:15:06] Raw response: {"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":{"config":{},"api_extensions":["storage_zfs_remove_snapshots","container_host_shutdown_timeout","container_syscall_filtering","auth_pki","container_last_used_at","etag","patch","usb_devices","https_allowed_credentials","image_compression_algorithm","directory_manipulation","container_cpu_time","storage_zfs_use_refquota","storage_lvm_mount_options","network","profile_usedby","container_push","container_exec_recording","certificate_update","container_exec_signal_handling","gpu_devices","container_image_properties","migration_progress","id_map","network_firewall_filtering","network_routes","storage","file_delete","file_append","network_dhcp_expiry","storage_lvm_vg_rename","storage_lvm_thinpool_rename","network_vlan","image_create_aliases","container_stateless_copy"],"api_status":"stable","api_version":"1.0","auth":"trusted","public":false,"environment":{"addresses":[],"architectures":["x86_64","i686"],"certificate":"-----BEGIN CERTIFICATE-----\nMIIFPTCCAyWgAwIBAgIQJlQRzEuSp2/pd7slP9k6ajANBgkqhkiG9w0BAQsFADAz\nMRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMRMwEQYDVQQDDApyb290QGx4\nZDAxMB4XDTE3MDYxOTA1NDc1NFoXDTI3MDYxNzA1NDc1NFowMzEcMBoGA1UEChMT\nbGludXhjb250YWluZXJzLm9yZzETMBEGA1UEAwwKcm9vdEBseGQwMTCCAiIwDQYJ\nKoZIhvcNAQEBBQADggIPADCCAgoCggIBAMILH6rWNyKjcm8KvWH+q8vFeOKNNZu+\nc0sDSjyLcjWyUNH6Dp4PXsB07qSMjaTxbYbq/k1P+n4LP3PMB610sbda9NNTXd0c\nuM0aQrRXCe//26h6UscjptArREQfoo/R+fEMteKh0gXtmdUA8H8G1FBwKGgXRmW3\nO9Ij8XmHJZIOQjnr3drAC+NPwyKzSw9hu2Ns7O0NEi5Aign2OApY4mdJRg377gNu\nHwYJUPMy51a337vTeNJ26EiBDALe1l97ImABFKyX+sZl9QqCzvd5uAYxcia3Ut8G\nZQR5t5nPunsDFWFfnGFRHXLEXwZoZ0Ck/6k1E5XwJ1o4x/KVOzwo3tBNPDrHMQqw\n4Ffk3TUvI7BsFSmRrVzKkZo096VvlAKzzWg6S2hg6ijPcIWFeZ1886/3IsfIpNDf\nlsdzRITK/DOlAMa9pHUd+BAwZIZcgbghCA8zqdyng8YpB8eijFHXR8ilyvMNS+mC\nWMpChehvOGKZRsSX/fKGgSaWmogAhYV9GlE6O4tmpgyf/pR41b3kLLWImvDH2GFi\nTh+k48aIqZFx8/9I5MJ6z5yGoYBf9N4gQxt/3WqJ44zGjuwm+qsexX8khzdpApQr\noAqYg48mhK5/YlZIksCk+sUQxV5FXYdsol2ISegEjpC6TQ0rlqBfOlLCJpkoueFm\n5xVo03mgcOh5AgMBAAGjTTBLMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggr\nBgEFBQcDATAMBgNVHRMBAf8EAjAAMBYGA1UdEQQPMA2CBWx4ZDAxhwQK/wAFMA0G\nCSqGSIb3DQEBCwUAA4ICAQAoo6DZHQujHfjMajgSRSyN7lUyz7vleAUU2Yk/aOx9\ng83+mcsfvRpgfpnlHVRzBWUfszgqgb9JiS79oZWiRB9eJk2dszSQFig4/x1yetkM\nJUCxcabwLhdHbk+WsrVOG9lC4psecK2ouC+rMZtZLp705fhVKB20Z7i2udD5AAwW\n+l9mKzeVddHkl/jMzYrMTvAr5XamK1v9I85lo8a6b4e/KZXFc4QNRvtCff9uvYE9\nx2ByhNvGMSk4V0asS5HJVidC7TyFtn4vuoXk96ZI3ySfXP3C7xA+77taQC59Ng/n\nVy+YH809xyb2QqOJVmphiDxE0QdpTcuWYI2lveDrBxStkxoUw8lRum3kiMwqqR18\n9kFknVhiDEGoe56uN/z8+nUSpJyybRcBNqcdsUdtwn1HC0Cn3Xa0xptJviQs//AO\ndjZgMWneq/f82bGmkh8/xReXWz7yDxfeV8MJIYFLjcgzvDtioujbiaEU5nRtS9SC\nzo7whcxl+s2pJnq0H+//KJ1xQprRxSECeKZnFOskn6b/Ux14vMwp1QLWLDhzKGg2\nnSGTORi859g3EOCbi6K5Fl0gkxVw3IFDlbbMTEHiejgK1EQdBt8+QKRSymEcTZi8\nl9WvynDXuk2rqnfnHw/K6JeV0B1/2O+w8r1a93W8Teq88CJJGgZUKHa9jnB8CVe5\nYA==\n-----END CERTIFICATE-----\n","certificate_fingerprint":"","driver":"lxc","driver_version":"2.0.7","kernel":"Linux","kernel_architecture":"x86_64","kernel_version":"4.10.0-24-generic","server":"lxd","server_pid":1369,"server_version":"2.12","storage":"zfs","storage_version":"0.6.5.9-2"}}}

DBUG[06-23|14:15:06] POST {"devices":{},"name":"ca","profiles":["default"],"source":{"certificate":"","fingerprint":"default","mode":"pull","protocol":"simplestreams","server":"https://cloud-images.ubuntu.com/releases","type":"image"}}
 to http://unix.socket/1.0/containers
DBUG[06-23|14:15:06] Raw response: {"type":"async","status":"Operation created","status_code":100,"operation":"/1.0/operations/0623de12-1340-4f2a-865d-7da0daa9f7f6","error_code":0,"error":"","metadata":{"id":"0623de12-1340-4f2a-865d-7da0daa9f7f6","class":"task","created_at":"2017-06-23T14:15:06.025390269-04:00","updated_at":"2017-06-23T14:15:06.025390269-04:00","status":"Running","status_code":103,"resources":{"containers":["/1.0/containers/ca"]},"metadata":null,"may_cancel":false,"err":""}}

Creating ca
DBUG[06-23|14:15:08] Raw response: {"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":{"id":"0623de12-1340-4f2a-865d-7da0daa9f7f6","class":"task","created_at":"2017-06-23T14:15:06.025390269-04:00","updated_at":"2017-06-23T14:15:06.025390269-04:00","status":"Success","status_code":200,"resources":{"containers":["/1.0/containers/ca"]},"metadata":null,"may_cancel":false,"err":""}}

DBUG[06-23|14:15:08] Raw response: {"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":{"architecture":"x86_64","config":{"image.architecture":"amd64","image.description":"ubuntu 16.04 LTS amd64 (release) (20170619.1)","image.label":"release","image.os":"ubuntu","image.release":"xenial","image.serial":"20170619.1","image.version":"16.04","volatile.apply_template":"create","volatile.base_image":"7a7ff654cbd8f5f09bec03aa19d8d7d92649127d18659036a963b1ea63f90d25","volatile.idmap.base":"0","volatile.idmap.next":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]","volatile.last_state.idmap":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]"},"devices":{},"ephemeral":false,"profiles":["default"],"created_at":"2017-06-23T14:15:06-04:00","expanded_config":{"image.architecture":"amd64","image.description":"ubuntu 16.04 LTS amd64 (release) (20170619.1)","image.label":"release","image.os":"ubuntu","image.release":"xenial","image.serial":"20170619.1","image.version":"16.04","raw.lxc":"lxc.id_map =\nlxc.id_map = u 0 100000 200\nlxc.id_map = g 0 100000 200\nlxc.id_map = u 200 200 1800\nlxc.id_map = g 200 200 1800\nlxc.id_map = u 2000 102000 63536\nlxc.id_map = g 2000 102000 63536\n","user.vendor-data":"#cloud-config\n\n# Set timezone\ntimezone: US/Eastern\n\n# User setup\nusers:\n - name: jfgibbins\n   uid: \"1000\"\n   sudo: ['ALL=(ALL) NOPASSWD:ALL']\n   groups: sudo\n   shell: /bin/bash\n - name: jfryer\n   uid: \"1001\"\n   sudo: ['ALL=(ALL) NOPASSWD:ALL']\n   groups: sudo\n   shell: /bin/bash\n\n# Run commands\nruncmd:\n# - dpkg -i --force-confdef /home/jfgibbins/htpc/apps/puppet-agent_1.9.3-1xenial_amd64.deb\n\n# Output\noutput: {all: '| tee -a /var/log/cloud-init-output.log'}\n\n# Final_message\nfinal_message: \"The system is finally up, after $UPTIME seconds\"\n\n","volatile.apply_template":"create","volatile.base_image":"7a7ff654cbd8f5f09bec03aa19d8d7d92649127d18659036a963b1ea63f90d25","volatile.idmap.base":"0","volatile.idmap.next":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]","volatile.last_state.idmap":"[{\"Isuid\":true,\"Isgid\":false,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536},{\"Isuid\":false,\"Isgid\":true,\"Hostid\":100000,\"Nsid\":0,\"Maprange\":65536}]"},"expanded_devices":{"jfgibbins":{"path":"/home/jfgibbins","source":"/home/jfgibbins","type":"disk"},"jfryer":{"path":"/home/jfryer","source":"/home/jfryer","type":"disk"},"root":{"path":"/","pool":"lxd","type":"disk"},"tmp":{"path":"/tmp","source":"/tmp","type":"disk"}},"name":"ca","stateful":false,"status":"Stopped","status_code":102,"last_used_at":"1969-12-31T19:00:00-05:00"}}

The container you are starting doesn't have any network attached to it.
  To create a new network, use: lxc network create
  To attach a network to a container, use: lxc network attach

Starting ca
DBUG[06-23|14:15:08] PUT {"action":"start","force":false,"stateful":false,"timeout":-1}
 to http://unix.socket/1.0/containers/ca/state
DBUG[06-23|14:15:08] Raw response: {"type":"async","status":"Operation created","status_code":100,"operation":"/1.0/operations/621bd739-d76d-4e15-9295-ce041b6a5193","error_code":0,"error":"","metadata":{"id":"621bd739-d76d-4e15-9295-ce041b6a5193","class":"task","created_at":"2017-06-23T14:15:08.351313528-04:00","updated_at":"2017-06-23T14:15:08.351313528-04:00","status":"Running","status_code":103,"resources":{"containers":["/1.0/containers/ca"]},"metadata":null,"may_cancel":false,"err":""}}

**Comment by brauner** [comment]
Can you try and reproduce with `2.14`?

**Comment by stgraber** [comment]
Marking as incomplete as the report doesn't show any actual error or failure in any of the provided logs or output.

**Comment by jfgibbins** [comment]
thank brauner, I spun up 2.14 on a test box, ca creates fine on it,and not sure what changed between the versions or caused just ca to not function.   I'll have to update prod, and sorry stgraber, lack of error messages in log, because process never completed.  running lxc start ca never finished or produced anything, and provided all I knew how to find, any other suggestions on debugging/logging would be appreciated, as I pretty much have found all the other errors since I've been playing with lxd the last 9 months

**Comment by stgraber** [comment]
Ah, so it's a hang at startup. That could be our favorite lxc-monitord bug at play.

Can you reproduce this issue reliably or is intermittent? If the latter, it's most likely the monitord isuse. You can unstick "lxc start" in such cases by running "pkill -9 lxc-monitord".

The next LXC release should have this resolved by having us remove the dependency on the monitord code entirely.

**Comment by jfgibbins** [comment]
Like 99% of the time.  That's what was messing me up.  I thought I was screwing up with my terraform implementation, cause on rare occasion it would work.  But then I dropped to just the basic lxc launch etc, and it failed pretty much every time I tested it, hence why I went to adding as issue. Racked my brain last night screwing with this and terraform.  Even recompiled terraform lxd-provider from github for the latest updates.  And thanks for the kill command, that will save me some copy/paste when I hang an lxc start, which I've done on occasion when my idmap isn't aligned with my subuid/gid settings, but that's another issue, maybe I should add that one too to the list, sorry for the list of issues adding today
